library/check_crypto_config.h

```diff
@@ -28,6 +28,18 @@
 #ifndef MBEDTLS_CHECK_CRYPTO_CONFIG_H
 #define MBEDTLS_CHECK_CRYPTO_CONFIG_H
 
+#if defined(PSA_WANT_ALG_CCM) && \
+    !( defined(PSA_WANT_KEY_TYPE_AES) || \
+       defined(PSA_WANT_KEY_TYPE_CAMELLIA) )
+#error "PSA_WANT_ALG_CCM defined, but not all prerequisites"
+#endif
+
+#if defined(PSA_WANT_ALG_CMAC) && \
```

Comment:
This checks that if CMAC is requested then we have at least one block cipher to go with it. If we add support for another block cipher in CMAC we will add to this list.

Reply:
I'm just wondering about the case if eg. someone wants to build mbedtls with a driver that supports eg. Camellia-CMAC.

I think that's a valid configuration, that should be possible, even if the core of MbedTLS doesn't support that combination.

Reply:
Sorry, I missed the point of built-in/accelerator I think.

Reply:
Agreed thus I think the idea is to validate here from the point of view of the PSA spec thus rather than removing it I think we should rather add CAMELLIA here, right?